IPs/RSKIP502.md

```diff
@@ -0,0 +1,175 @@
+- 0: Success – The Union Bridge contract address was successfully updated.
+- -1: Unauthorized caller – The caller does not have the necessary permissions to update the contract address.
+- -10: Generic error – A non-specific error occurred while processing the request.
+
+**Environment restriction:**
+
+This method will be only enabled for testnet and regtest environments.
+It will be disabled on mainnet to prevent unauthorized updates.
+
+### 3. Authorizer for changing the Union Bridge contract address
+
+An authorizer key will be created to allow the update of the Union Bridge contract address. This key will be similar to the whitelist authorizer [2], with a single key being sufficient for this purpose. No voting or multisig will be needed since this will only be relevant on testnet.
+
+### 4. Union Bridge locking cap
+
+A locking cap will be implemented to control the maximum amount of RBTC that can be transferred to the Union Bridge. This cap helps mitigate risks by preventing excessive transfers beyond the system's control limits.
+
+In order to integrate the locking cap logic with the Union Bridge, new methods need to be added to the existing Bridge contract to enable querying and updating the locking cap. Based on RSKIP134 [3], which provides a standard mechanism for managing caps in smart contracts, new methods are defined.
+
+#### 4.1 Method to get the current locking cap
+
+A method will be added to allow anyone to query the current locking cap. This ensures transparency regarding the maximum allowable amount that can be transferred to the Union Bridge.
+
+**Method signature:**
+
+```
+function getUnionBridgeLockingCap() public view returns (uint256);
+```
+
+#### 4.2 Method to adjust the locking cap
+
+This method will allow authorized accounts to adjust the locking cap. The cap can only be increased, and the increase should be subject to a predefined maximum (e.g., doubling the current cap). The purpose of the adjustment method is to ensure that the cap can scale with demand, but only in a controlled way.
+
+**Method signature:**
+
+```
+function increaseUnionBridgeLockingCap(uint256 newCap) public returns (int256);
+```
+
+**Response codes:**
+
+- 0: Success – The locking cap has been adjusted successfully.
+-1: Unauthorized caller – Only authorized entities can adjust the locking cap.
+-2: Invalid value – The specified cap value is invalid (e.g., less than the current cap or excessive).
+- -10: Generic error – A non-specific error occurred while processing the request.
+
+### 5. Method for Transferring RBTC to the Union Bridge
+
+A new method will be added to the Bridge contract to allow the transfer of RBTC to the Union Bridge. The method will have the following restrictions:
+
+- Only callable by the Union Bridge contract.
+- The total requested amount (current request + previously requested amounts) must not exceed the current locking cap.
+
+**Method Signature:** 
+```
+function requestRBTC(uint256 amountInWeis) public returns
```

Comment:
The method should be called requestUnionRBTC() to avoid confusion and match a specific method used to release of btc

Reply:
Also we could add a destination address so we can send tokens directly. That way we will save gas, avoiding  having to send the tokens to the union bridge and then to the user.
 Should end up looking something like this
 
```suggestion
function requestUnionRBTC(address to, uint256 amountInWeis) public returns
```

Reply:
I like Peter's idea about transferring the rbtc directly but we must not forget the bridge doesn't do a transfer per but instead it changes the balance, therefore if the recipient is a contract it won't execute any code